### PR TITLE
[NEW] App empty states component, category filter and empty states error variation implementations

### PR DESCRIPTION
**Pull request by rique223**
## Proposed changes (including videos or screenshots)

Created and implemented the category filters component:
Demo gif:
categories_filter_demo

Created and implemented the empty states(States on fuselage) component:
Demo gif:
empty_states_demo

Implemented a variations system for the empty states component and created a error message for network outage:
Demo gif:
empty_states_variation_demo

### CODE REVIEW

File: client/lib/asyncState/AsyncState.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import { AsyncStatePhase } from './AsyncStatePhase';
 
 export type AsyncState<T> =
-	| { phase: AsyncStatePhase.LOADING; value: undefined; error: undefined }
+	| {
+			phase: AsyncStatePhase.LOADING;
+			value: undefined;
+			error: undefined;
+	  }
```
Comment (geekgonecrazy) [inline client/lib/asyncState/AsyncState.ts:8]: For consistency I wonder should this stay one line?

File: client/views/admin/apps/AppsRoute.tsx [hunk 2]
```diff
@@ -39,17 +39,17 @@ const AppsRoute = () => {
...
 	}, [canViewAppsAndMarketplace, isAppsEngineEnabled, appsWhatIsItRoute]);
 
 	const context = useRouteParameter('context');
 
-	const isMarketPlace = !context;
+	const isMarketplace = !context;
 
 	const id = useRouteParameter('id');
-	const version = useRouteParameter('version');
+	// const version = useRouteParameter('version');
```
Comment (geekgonecrazy) [inline client/views/admin/apps/AppsRoute.tsx:52]: Hm.. what was version being used for before?

File: client/views/admin/apps/hooks/useCategories.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+// 			{ id: '17', label: 'Project Management', checked: false },
+// 			{ id: '18', label: 'Sales', checked: false },
+// 			{ id: '19', label: 'Security & Compliance', checked: false },
+// 			{ id: '20', label: 'Social & Fun', checked: false },
+// 			{ id: '21', label: 'Team Culture', checked: false },
+// 			{ id: '22', label: 'Travel', checked: false },
+// 			{ id: '23', label: 'Voice & Video', checked: false },
+// 		],
+// 	},
+// ];
```
Comment (ggazzo) [inline client/views/admin/apps/hooks/useCategories.ts:47]:
```suggestion
```

File: client/views/admin/apps/hooks/useCategories.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+				label: currentCategory.title,
+				checked: false,
+			}));
+
+			setCategories([
+				{
+					items: [
+						{
+							id: 'all',
+							label: 'All categories',
```
Comment (ggazzo) [inline client/views/admin/apps/hooks/useCategories.ts:80]: missing translation

File: client/views/admin/apps/hooks/useCategories.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+						},
+					],
+				},
+				{
+					label: 'Filter by Category',
```
Comment (ggazzo) [inline client/views/admin/apps/hooks/useCategories.ts:85]: missing translation

File: client/views/admin/apps/AppsRoute.tsx (hunk 2 above)
Comment (ggazzo) [reply client/views/admin/apps/AppsRoute.tsx:52]: it was used to be passed to a inner component, but now we are performing `useRouteParameter('version')` there

Comment (ggazzo) [inline client/views/admin/apps/AppsRoute.tsx:52]:
```suggestion
```

File: app/theme/client/imports/general/base_old.css [hunk 6]
```diff
@@ -1059,7 +1059,7 @@
 		display: flex;
 		align-items: center;
 
-		& input[type=file] {
+		& input[type='file'] {
 			position: absolute !important;
```
Comment (ujorgeleite) [inline app/theme/client/imports/general/base_old.css:1063]: if is possible use specific selectors, to not use !important

File: client/sidebar/header/actions/Search.js [hunk 7]
```diff
@@ -9,12 +9,12 @@ const Search = (props) => {
 	const [searchOpen, setSearchOpen] = useState(false);
 
 	const ref = useRef();
-
 	const handleCloseSearch = useMutableCallback(() => {
 		setSearchOpen(false);
 		// viewRef.current && Blaze.remove(viewRef.current);
```
Comment (ujorgeleite) [inline client/sidebar/header/actions/Search.js:14]: Remove commented code.

File: client/views/admin/apps/AppDetailsPage.tsx [hunk 8]
```diff
@@ -62,10 +64,10 @@ function AppDetailsPage({ id }) {
 			</Page.Header>
 			<Page.ScrollableContentWithShadow>
 				<Box maxWidth='x600' w='full' alignSelf='center'>
-					{isLoading && <LoadingDetails />}
-					{!isLoading && (
+					{!data && <LoadingDetails />}
+					{data && (
 						<>
-							<AppDetailsPageContent data={data} />
+							<AppDetailsPageContent app={data} />
```
Comment (ujorgeleite) [inline client/views/admin/apps/AppDetailsPage.tsx:70]: I think that lines 67 to 74 can be improved by breaking these validations into small functions to make this easy to be read.

File: client/views/admin/apps/AppDetailsPageContent.tsx [hunk 9]
```diff
@@ -46,26 +46,26 @@ const AppDetailsPageContent: FC<AppDetailsPageContentProps> = ({ data }) => {
...
 						</Box>
-						{installed && <AppMenu app={data} />}
+						{installed && <AppMenu app={app} />}
 					</Box>
 				</Box>
 			</Box>
 			<Divider />
 
-			{data.licenseValidation && (
+			{app.licenseValidation && (
```
Comment (ujorgeleite) [inline client/views/admin/apps/AppDetailsPageContent.tsx:60]: from line 60 to line 70 can be improved by breaking the validations into small functions to make easy the read around this code.

**Review by ujorgeleite — CHANGES_REQUESTED** [review]

**Review by ujorgeleite — APPROVED** [review]

**Pull request merged**